db/merge_helper.cc

```diff
@@ -56,13 +56,12 @@ MergeHelper::MergeHelper(Env* env, const Comparator* user_comparator,
   }
 }
 
-Status MergeHelper::TimedFullMerge(const MergeOperator* merge_operator,
-                                   const Slice& key, const Slice* value,
-                                   const std::vector<Slice>& operands,
-                                   std::string* result, Logger* logger,
-                                   Statistics* statistics, SystemClock* clock,
-                                   Slice* result_operand,
-                                   bool update_num_ops_stats) {
+Status MergeHelper::TimedFullMerge(
+    const MergeOperator* merge_operator, const Slice& key, const Slice* value,
+    const std::vector<Slice>& operands, std::string* result, Logger* logger,
+    Statistics* statistics, SystemClock* clock, Slice* result_operand,
+    bool update_num_ops_stats,
+    MergeOperator::OpFailureScope* op_failure_scope) {
```

Comment:
I think now we follow updated Google C++ style and output parameter uses reference. And if we indicate a parameter not needed by caller, consider to try std::optional, although I think an enum is cheap so we can afford for the callers to always have it.

Reply:
The current text says "Use non-const pointers to represent optional outputs" (https://google.github.io/styleguide/cppguide.html#Inputs_and_Outputs)

Reply:
>  I think an enum is cheap so we can afford for the callers to always have it

It is true it is cheap, but I figured if there are many callers who won't even look at the result, it might make sense still for it to be optional. So I tried leaving it as a pointer for now